python/arcticdb/version_store/library.py

```diff
@@ -900,22 +899,28 @@ def finalize_staged_data(
         mode: Optional[StagedDataFinalizeMethod] = StagedDataFinalizeMethod.WRITE,
         prune_previous_versions: bool = False,
         metadata: Any = None,
+        validate_index=True,
     ):
         """
-        Finalises staged data, making it available for reads.
+        Finalizes staged data, making it available for reads.
 
         Parameters
         ----------
         symbol : `str`
             Symbol to finalize data for.
 
         mode : `StagedDataFinalizeMethod`, default=StagedDataFinalizeMethod.WRITE
-            Finalise mode. Valid options are WRITE or APPEND. Write collects the staged data and writes them to a
-            new timeseries. Append collects the staged data and appends them to the latest version.
+            Finalize mode. Valid options are WRITE or APPEND. Write collects the staged data and writes them to a
+            new version. Append collects the staged data and appends them to the latest version.
         prune_previous_versions: bool, default=False
             Removes previous (non-snapshotted) versions from the database.
         metadata : Any, default=None
             Optional metadata to persist along with the symbol.
+        validate_index: bool, default=True
```

Comment:
It would be good to describe why anyone would want to set this to False. Is it just for a performance boost?

Reply:
No, it is for if you want to allow unordered data. It is following the same pattern as the docstrings for `write` etc, but the phrasing has to be a bit more complex because you're creating the new version from incompletes.